wandb/wandb_config.py

```diff
@@ -172,21 +172,26 @@ def update(self, params, allow_val_change=False):
         if not isinstance(params, dict):
             # Handle some cases where params is not a dictionary
             # by trying to convert it into a dictionary
+            if "__name__" in dir(params) and params.__name__ in ('tensorflow.python.platform.flags', 'absl.flags'):
+                params = params.FLAGS
 
             if not hasattr(params, '__dict__'):
                 raise TypeError(
                     "config must be a dict or have a __dict__ attribute.")
-            if "__flags" in vars(params):
-                # for older tensorflow flags (pre 1.4)
-                if not '__parsed' in vars(params):
-                    params._parse_flags()
-                params = vars(params)['__flags']
-            elif "__wrapped" in vars(params):
-                # newer tensorflow flags (post 1.4) uses absl.flags in a wrapper
-                params = {name: params[name].value for name in dir(params)}
-            else:
-                # params is a Namespace object (argparse)
-                # or something else
+            try:
```

Comment:
This is super gross, don't know how to ensure I can call `.__module__` as it's not in vars.  I needed this to run something from the tensorflow/models repo as it was using absl.flags directly and didn't have a `__wrapped` attribute